app/models/concerns/statusable.rb

```diff
@@ -27,4 +29,11 @@ def contact_made?
     end
     false
   end
+
+  def days_since_last_call
+    return 0 if calls.blank?
```

Comment:
A couple notes about this:

* If a patient doesn't have any calls this will choke, so we add the safety condition up top
* `calls.sort_by...` is replaced with `last_call`, but the core of your flow here is still much intact